crates/l2/sequencer/block_producer/payload_builder.rs

```diff
@@ -0,0 +1,274 @@
+                    MetricsTxStatus::Succeeded,
+                    MetricsTxType(head_tx.tx_type())
+                ));
+                receipt
+            }
+            // Ignore following txs from sender
+            Err(e) => {
+                debug!("Failed to execute transaction: {}, {e}", tx_hash);
+                metrics!(METRICS_TX.inc_tx_with_status_and_type(
+                    MetricsTxStatus::Failed,
+                    MetricsTxType(head_tx.tx_type())
+                ));
+                txs.pop();
+                continue;
+            }
+        };
+        // Add transaction to block
+        debug!("Adding transaction: {} to payload", tx_hash);
+        context.payload.body.transactions.push(head_tx.into());
+        // Save receipt for hash calculation
+        context.receipts.push(receipt);
+    }
+    Ok(())
+}
+
+/// Calculates the size of the current `StateDiff` of the block.
+/// If the current size exceeds the blob size limit, returns `Ok(false)`.
+/// If there is still space in the blob, returns `Ok(true)`.
+fn check_state_diff_size(
+    acc_withdrawals_size: &mut usize,
+    acc_deposits_size: &mut usize,
+    tx: Transaction,
+    receipt: &Receipt,
+    context: &mut PayloadBuildContext,
+    accounts_info_cache: &mut HashMap<Address, Option<AccountInfo>>,
+) -> Result<bool, BlockProducerError> {
+    if is_withdrawal_l2(&tx, receipt)? {
+        *acc_withdrawals_size += L2_WITHDRAWAL_SIZE;
+    }
+    if is_deposit_l2(&tx) {
+        *acc_deposits_size += L2_DEPOSIT_SIZE;
+    }
+    let modified_accounts_size = calc_modified_accounts_size(context, accounts_info_cache)?;
+
+    let current_state_diff_size = 1 /* version (u8) */ + HEADER_FIELDS_SIZE + *acc_withdrawals_size + *acc_deposits_size + modified_accounts_size;
+
+    if current_state_diff_size > SAFE_BYTES_PER_BLOB {
+        // Restore the withdrawals and deposits counters.
+        if is_withdrawal_l2(&tx, receipt)? {
+            *acc_withdrawals_size -= L2_WITHDRAWAL_SIZE;
+        }
+        if is_deposit_l2(&tx) {
+            *acc_deposits_size -= L2_DEPOSIT_SIZE;
+        }
+        debug!(
+            "Blob size limit exceeded. current_state_diff_size: {}",
+            current_state_diff_size
+        );
+        return Ok(false);
+    }
+    Ok(true)
+}
+
+fn calc_modified_accounts_size(
+    context: &mut PayloadBuildContext,
+    accounts_info_cache: &mut HashMap<Address, Option<AccountInfo>>,
+) -> Result<usize, BlockProducerError> {
+    let mut modified_accounts_size: usize = 2; // modified_accounts_len(u16)
+
+    // We use a temporary_context because revm mutates it in `get_state_transitions`
+    // TODO: remove when we stop using revm
+    let mut temporary_context = context.clone();
+    let account_updates = temporary_context
+        .vm
+        .get_state_transitions(context.payload.header.parent_hash)?;
+    for account_update in account_updates {
+        modified_accounts_size += 1 + 20; // r#type(u8) + address(H160)
```

Comment:
It wasn't hard to follow, but i feel that having the comments like this would help:
```suggestion
        modified_accounts_size += 1 + 20; // 1byte + 20bytes // r#type(u8) + address(H160)
```

Indicating the bytes needed and then the variables.

Reply:
Addressed in [651ed90](https://github.com/lambdaclass/ethrex/pull/2292/commits/651ed903f65b4ed9ea56cacfd09cb5ef58eb30f0)!